lib/user_repository.rb

```diff
@@ -0,0 +1,58 @@
+require_relative "user"
+require "bcrypt"
```

Comment:
Overall, I really like your layout of the code, all your methods are short and very readable, and your variable names are well chosen. Nice work.